src/main/java/net/minecraftforge/fmp/multipart/IMultipart.java

```diff
@@ -0,0 +1,358 @@
+package net.minecraftforge.fmp.multipart;
+
+import java.util.List;
+import java.util.Random;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.BlockStateContainer;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.color.IBlockColor;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.common.property.ExtendedBlockState;
+import net.minecraftforge.common.util.RayTraceUtils;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.fmp.client.multipart.AdvancedParticleManager;
+import net.minecraftforge.fmp.client.multipart.FastMSR;
+import net.minecraftforge.fmp.client.multipart.MultipartSpecialRenderer;
+import net.minecraftforge.fmp.client.multipart.MultipartStateMapper;
+import net.minecraftforge.fmp.multipart.IPartFactory.IAdvancedPartFactory;
+
+/**
+ * Interface that represents a multipart. This has all the methods required for a multipart to work properly.<br/>
+ * For a default implementation of most of these methods and some helpers, you can extend {@link Multipart} directly.
+ *
+ * @see Multipart
+ */
+public interface IMultipart extends ICapabilityProvider
```

Comment:
This is a instance based interface? How would this be used? Looks mostly like a wrapper around Block...